### DISCUSSION TITLE
Events: filter events based on variable (like {{event_file_name}} before passing to shell

### DISCUSSION THREAD

**Discussion by thomasvs** [Taitava/obsidian-shellcommands#403] [Ideas - on hold]
I have a command that runs an external command to update a panel message whenever I change my "My Daily Reminders.md" file.  I use the File Modified event, but it runs on any file modified, which means obsidian is constantly spawning shells as I type.  It would be great to avoid that by adding a filter to events based on the variable content.

**Comment by Taitava** [comment]
Hi,

thank you for your suggestion! 🙂 

I think there are two solutions that could work together:
1) In your shell command, you can use an if block to only call the external command if the modified file was `My Daily Reminders.md`. I'm not certain if your shell is Bash, but here's a Bash example anyway:
    ```bash
    if [ {{event_file_name}} = "My Daily Reminders.md" ]
    then
        echo "Call the external command here."
    fi
    ```
2) To reduce the times shells are spawned unnecessarily when editing files, there is a feature called [event debouncing](https://publish.obsidian.md/shellcommands/Events/Events+-+debouncing), which is still under development, but can be [beta tested](https://github.com/Taitava/obsidian-shellcommands/discussions/391) already. It prevents executing a shell command if an event has just executed it a short time ago.

---

As for the actual feature request about bringing conditions to event execution, I have thought about the idea in the past (but probably never written about it) and came up to a conclusion it's not so beneficial compared to the effort it would require:
- Conditions can be made in the shell command itself, and they support flexible syntax for advanced conditions (e.g. and/or operators with parenthesed conditions, and conditions based on an external program's result) that would be relatively hard to implement in the SC plugin.
- Having conditions in the SC plugin would create a competing feature which would not bring enough benefits to justify the development effort.

This is my stand at the moment. However, I do appreciate that you brought up this suggestion! 🙂
